src/errorhandler/errorhandler.cpp

```diff
@@ -0,0 +1,87 @@
+#include "errorhandler.hpp"
+
+sErrorStructure cErrorHandler::ErrorStructure[E_NUMBER_OF_ERRORS] =
+{
+// Error number         |       Error active        |       Error enable        |       Reaction
+{  E_FREE_TO_USE,              false,                      E_ERROR_ENABLED,            NONE_REACT}
+};
+
+
+cErrorHandler::cErrorHandler(void) :
+    clearErrorTable(false)
+{
+    for(ui8 loopIndex = 0U; loopIndex < errorTableSize; loopIndex++)
+    {
+        errorTable[loopIndex] = E_NO_ERROR;
+    }
+}
+
+void cErrorHandler::SetErrorActive(eErrors arg_errorNumber)
+{
+    ErrorStructure[static_cast<ui8>(arg_errorNumber)].active = true;
+}
+
+void cErrorHandler::SetErrorDisactive(eErrors arg_errorNumber)
+{
+    ErrorStructure[static_cast<ui8>(arg_errorNumber)].active = false;
+}
+
+void cErrorHandler::SetError(eErrors arg_errorNumber, bool arg_isErrorActive)
+{
+    ErrorStructure[static_cast<ui8>(arg_errorNumber)].active = arg_isErrorActive;
+}
+
+void cErrorHandler::Cyclic(void)
+{
+    for(ui8 errorIndex = 0U; errorIndex < E_NUMBER_OF_ERRORS; errorIndex++)
+    {
+        if(WouldErrorBeProccessed(errorIndex))
+        {
+            ProccessError(errorIndex);
+            ErrorStructure[errorIndex].reaction->ErrorReaction();
+        }
+    }
+
+    if (clearErrorTable)
+    {
+        ClearErrorTable();
+    }
+}
+
+bool cErrorHandler::WouldErrorBeProccessed(const ui8 arg_errorIndex)
+{
+    return ((ErrorStructure[arg_errorIndex].enable == E_ERROR_ENABLED) && (ErrorStructure[arg_errorIndex].active == true));
+}
+
+void cErrorHandler::ProccessError(const ui8 arg_errorIndex)
+{
+    bool errorAlreadyInTable = false;
+    ui8 numberOfFirstFreePlace = 255U;
+
+    for(ui8 errorTableIndex = 0U; errorTableIndex < errorTableSize; errorTableIndex++)
```

Comment:
For every active error you are scanning error table if the error is already here. And for every active error you are searching for last spot in the table. This is not so effective. Imagine you would have tousands of errors - that is lots of instructions.
I would suggest to have one variable with last position in table and for every error code flag if error is already in table. When error has lots of flags you can group it to bit field to save some space.